spark/src/main/kotlin/com/adevinta/spark/tools/modifiers/Layout.kt

```diff
@@ -53,18 +48,32 @@ import androidx.compose.ui.unit.Dp
  * @param horizontalPadding The horizontal padding to ignore from the parent
  */
 @OptIn(ExperimentalComposeUiApi::class)
-public fun Modifier.ignoreParentHorizontalPadding(horizontalPadding: Dp): Modifier = this then modifierElementOf(
-    params = horizontalPadding,
-    create = { IgnoreParentHorizontalPaddingModifier(horizontalPadding) },
-    update = { it.horizontalPadding = horizontalPadding },
-    definitions = {
+public fun Modifier.ignoreParentHorizontalPadding(horizontalPadding: Dp): Modifier =
+    this then IgnoreParentHorizontalPaddingModifier(horizontalPadding)
+
+@OptIn(ExperimentalComposeUiApi::class)
+private data class IgnoreParentHorizontalPaddingModifier(
+    private val horizontalPadding: Dp,
+) : ModifierNodeElement<IgnoreParentHorizontalPaddingModifierNode>() {
+
+    override fun create(): IgnoreParentHorizontalPaddingModifierNode {
+        return IgnoreParentHorizontalPaddingModifierNode(horizontalPadding)
+    }
+
+    override fun update(node: IgnoreParentHorizontalPaddingModifierNode): IgnoreParentHorizontalPaddingModifierNode {
+        return node.apply {
+        }
+    }
```

Comment:
I think something is missing in the update method

```suggestion
    override fun create() = IgnoreParentHorizontalPaddingModifierNode(horizontalPadding)

    override fun update(node: IgnoreParentHorizontalPaddingModifierNode) = node.apply {
    }
```

Reply:
Indeed nice catch but I think we're required to specify the return type of the functions with our explicit api option